src/reuse/modules/flp/userLocks.ts

```diff
@@ -0,0 +1,111 @@
+"use strict";
+
+import { VerboseLoggerFactory } from "../../helper/verboseLogger";
+
+/**
+ * @class userLocks
+ * @memberof util
+ */
+
+export class UserLocks {
+  private vlf = new VerboseLoggerFactory("util", "userLocks");
+  private _srvUserLockInstance = null;
+  private _srvEshInstance = null;
+  private _options: { Client: string; UserId: string; SessionId: string } = {
+    Client: "",
+    UserId: "",
+    SessionId: ""
+  };
+  private async initUserLockService(user: string, password: string = "Welcome1!"): Promise<any> {
+    if (!this._srvUserLockInstance) {
+      const vl = this.vlf.initLog(await this.initUserLockService);
+      const params = browser.config.params;
+      if (params?.systemUrl && user && password) {
+        try {
+          this._srvUserLockInstance = await service.odata.init(`${params.systemUrl}/sap/opu/odata/sap/APS_IAM_MUS_V2UI`, user, password);
+        } catch (error) {
+          if (error instanceof Error) {
+            throw new Error(`Failed to initialize S4 User Lock Service: ${error.message}.`);
+          } else {
+            throw new Error("Failed to initialize S4 User Lock Service");
+          }
+        }
+      } else {
+        throw new Error("System URL is missing in the config file.");
+      }
+    }
+  }
+  private async initEnterpriseSearchHelpService(user: string, password: string = "Welcome1!"): Promise<any> {
+    if (!this._srvEshInstance) {
+      const vl = this.vlf.initLog(await this.initEnterpriseSearchHelpService);
+      const params = browser.config.params;
+      if (params?.systemUrl && user && password) {
+        try {
+          this._srvEshInstance = await service.odata.init(`${params.systemUrl}/sap/opu/odata/sap/ESH_SEARCH_SRV`, user, password);
+        } catch (error) {
+          if (error instanceof Error) {
+            throw new Error(`Failed to initialize Search Service: ${error.message}.`);
+          } else {
+            throw new Error("Failed to initialize Search Service");
+          }
+        }
+      } else {
+        throw new Error("System URL is missing in the config file.");
+      }
+    }
+  }
+
+  public async getLockEntries(user: string, password: string, technicalUserId?: string): Promise<number> {
+    const vl = this.vlf.initLog(await this.getLockEntries);
+    let resEsh = null;
+
+    if (technicalUserId === undefined) {
```

Comment:
the user can specify the userId so the ESH Service doesn't need to be initialised